nilearn/_utils/fixes/scikit_learn_validation.py

```diff
@@ -0,0 +1,400 @@
+# scikit-learn v0.17
+
+import numpy as np
+import warnings
+import scipy.sparse as sp
+
+
+class NotFittedError(ValueError, AttributeError):
+    """Exception class to raise if estimator is used before fitting
+    This class inherits from both ValueError and AttributeError to help with
+    exception handling and backward compatibility.
+    """
+
+
+def check_array(array, accept_sparse=None, dtype="numeric", order=None,
```

Comment:
atleast2d_or_csr => check_array 
Basically, check_X_y combines check_array (for X and y), and checks if their lengths are consistent